crates/solvers/src/domain/mod.rs

```diff
@@ -1,18 +1,9 @@
 //! Core solver engine logic.
 
 pub mod auction;
-pub mod baseline;
 pub mod dex;
 pub mod eth;
-pub mod legacy;
 pub mod liquidity;
-pub mod naive;
 pub mod order;
 pub mod solution;
-
-pub trait Solver: Send + Sync {
-    fn solve(
-        &self,
-        auction: auction::Auction,
-    ) -> futures::future::BoxFuture<Vec<solution::Solution>>;
-}
```

Comment:
Just moved this stuff into `domain::solver` to group all solvers and the `Solver` trait.